test/test_nn.py

```diff
@@ -3910,6 +3910,40 @@ def test_Conv2d_inconsistent_types_on_GPU_without_cudnn(self):
             # but it should work with the same type
             nn.functional.conv2d(inputs.float(), weights.float(), bias.float())
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
```

Comment:
please put the tests in TestNNDeviceType (it starts approx on line 9000), that's where new-style device-generic tests live. 
If you want to run only on cuda, theres @onlyCUDA decorator (you won't need skip if `not TEST_CUDA`). 
The preferred way is also to use @dtypes decorator rather than specify the default value of dtype argument.